ncm-download/src/main/perl/download.pm

```diff
@@ -133,7 +133,8 @@ sub Configure {
                 $success += $self->download(
                     file => $file,
                     href => $source,
-                    timeout => $inf->{timeout},
+                    timeout => $inf->{files}->{$f}->{timeout} || $inf->{timeout},
```

Comment:
if the timeout of a file `$inf->{files}->{$f}->{timeout}` is set to 0, this code will override it with whatever is set in as global timeout `$inf->{timeout}`.